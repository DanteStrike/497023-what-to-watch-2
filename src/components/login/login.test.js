import React from "react";
import renderer from "react-test-renderer";
import Login from "./login";
import {BrowserRouter as Router} from "react-router-dom";

it(`Component Login should render correctly`, () => {
  const component = renderer.create(
      <Router>
        <Login
          email={`email`}
          emailValidation={{
            isValid: true,
            msg: ``
          }}
          validateEmail={jest.fn()}
          onEmailChange={jest.fn()}
          password={`password`}
          passwordValidation={{
            isValid: true,
            msg: ``
          }}
          validatePassword={jest.fn()}
          onPasswordChange={jest.fn()}
          requestLogin={jest.fn()}
          toggleFormLock={jest.fn()}
          isSubmitting={false}
          serverError={{
            isError: false,
            target: ``,
            msg: ``
          }}
        />
      </Router>
  );

  expect(component).toMatchSnapshot();
});
