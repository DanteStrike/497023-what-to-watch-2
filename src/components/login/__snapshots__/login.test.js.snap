// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Login should render correctly 1`] = `
<div
  className="sign-in user-page__content"
  style={Object {}}
>
  <form
    action="#"
    className="sign-in__form"
    onSubmit={[Function]}
    style={Object {}}
  >
    <div
      className="sign-in__fields"
    >
      <div
        className="sign-in__field"
      >
        <input
          className="sign-in__input"
          disabled={false}
          id="user-email"
          name="user-email"
          onChange={[Function]}
          placeholder="Email address"
          type="email"
        />
        <label
          className="sign-in__label visually-hidden"
          htmlFor="user-email"
        >
          Email address
        </label>
      </div>
      <div
        className="sign-in__field"
      >
        <input
          className="sign-in__input"
          disabled={false}
          id="user-password"
          name="user-password"
          onChange={[Function]}
          placeholder="Password"
          type="password"
        />
        <label
          className="sign-in__label visually-hidden"
          htmlFor="user-password"
        >
          Password
        </label>
      </div>
    </div>
    <div
      className="sign-in__submit"
    >
      <button
        className="sign-in__btn"
        disabled={false}
        type="submit"
      >
        Sign in
      </button>
    </div>
  </form>
</div>
`;
